devto-draft.md

```diff
@@ -1 +1,393 @@
+
+With Vue, you use the `:key` and pass a value. In React it's more or less the same thing
+
+## Watchers and Observables
+
+I never quite understood design patterns that well. But the term "observable" finally made sense to me.
+
+Vue has something called watchers. This lets you call a getter everytime that variable changes.
+
+However, there's some caveats, because you could have watchers watching watchers. This causes circular referential issues.
+
+The Vue docs warn against this. I myself have accidentally done this using inputs to watch over other inputs.
+
+React does not have anything similar out of the box.
+
+## Computed properties
+
+This was a new concept to me, from the React world. In Vue, you can have variables that depend on other variables. Anytime a dependency changes, so does the computed property.
+
+It seems useful at first, but computed properties are only good if you aren't modifying it at all directly.
+
+When you start needing to, this is when you go away from computed properties entirely to the next item...
+
+## Methods
+
+Methods in Vue worked the same as in React. These methods were called through event-directives that extend from HTML DOM events. Examples include putting `@click` events, or `@input` if a user typed text in an input field.
+
+Methods are what you'll always fall back in Vue when computed properties and watchers don't fit your use-case.
+
+Methods are like any function in javascript. The return statement is totally optional, you can just use it to mutate data in the background _(not a good practice)_
+
+## Asynchronous Gotchas
+
+This is a gotcha that you find out later in Vue, and in React.
+In React, there is a built-in method called `setState()`. When `setState()` gets called, it gets handled asynchronously in the background.
+
+Vue is no different, you have a method called `Vue.nextTick()` that awaits the latest updated value to the data.
+
+One lesson I learned is that it's best to set local variables within your methods, and only mutate data attributes as needed.
+
+## Templates
+
+With Vue, there is something called `templates`. In your component, you have the ability to declare `<template>` slots. When that component gets called, you can inject HTML where those slots match.
+
+This has been immensely useful when working in libraries. I could inject functionality as I saw fit, to get the feature I needed.
+
+Templates do not get rendered as a `<div>` element in the DOM. React has a similar approach to templates called fragments, but the approach is limited to just the first element in your component.
+
+## Lifecycle methods
+
+Lifecycle methods describe what components do during their lifetime.
+
+Here are common ones you'll use in Vue:
+
+- mounted() - Similar to React's componentDidMount. This generally when making an Axios/fetch call to initialize data in your component from a backend.
```

Comment:
The sentence structure is not there. 

> This generally when making an Axios/fetch call to initialize data in your component from a backend.

Maybe: This method is triggered when making AJAX calls to initialize data in your component from a backend.